sources/models/src/modeled_types/shared.rs

```diff
@@ -647,6 +647,126 @@ mod test_sysctl_key {
+            !keyword.is_empty()
+                && keyword
+                    .chars()
+                    .all(|c| (c.is_ascii() && c.is_alphanumeric()) || c == '-' || c == '_')
+        });
+        ensure!(valid_key, error::InvalidBootconfigKeySnafu { input });
+        Ok(BootConfigKey {
+            inner: input.to_string(),
+        })
+    }
+}
+
+string_impls_for!(BootConfigKey, "BootConfigKey");
+
+#[cfg(test)]
+mod test_bootconfig_key {
+    use super::BootConfigKey;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_bootconfig_key() {
+        for ok in &[
+            "keyword1.keyword2",
+            "-keyword1.keyword2",
+            "_keyword.1.2.3",
+            "key_word",
+            "key-word",
+            "keyword1",
+            "keyword1-",
+            "keyword2_",
+        ] {
+            BootConfigKey::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn invalid_bootconfig_key() {
+        for err in &[
+            "", "①", ".", "!", "@", "#", "$", "%", "^", "&", "*", "(", ")", "\"", "'", "\\", "|",
+            "~", "`",
+        ] {
+            BootConfigKey::try_from(*err).unwrap_err();
+        }
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// BootConfigValue represents a string that is a valid Kernel boot config value; each value only
+/// contains printable characters or spaces except for delimiters such as semicolon, newline, comma,
+/// hash, and closing brace. These delimiters are only usable if the value itself is quoted with
+/// single-quotes or double-quotes. We can treat the value as if they're always quoted in the context
+/// of Bottlerocket settings.
+/// BootConfigValue stores the original string and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct BootConfigValue {
+    inner: String,
+}
+
+impl TryFrom<&str> for BootConfigValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, error::Error> {
+        ensure!(
+            input.chars().all(|c| c.is_ascii() && !c.is_ascii_control()),
+            error::InvalidBootconfigValueSnafu { input }
+        );
+        Ok(BootConfigValue {
+            inner: input.to_string(),
+        })
+    }
+}
+
+string_impls_for!(BootConfigValue, "BootConfigValue");
+
+#[cfg(test)]
+mod test_bootconfig_value {
+    use super::BootConfigValue;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_bootconfig_value() {
+        for ok in &[
+            "plain",
+            "@yogurt@",
+            "\"abc",
+            " !#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}",
+            "1",
+            "value1",
+            "hello.goodbye",
+            "",
+        ] {
+            BootConfigValue::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn invalid_bootconfig_value() {
+        for err in &["bottlerocket①", "💝", "Ï", "—"] {
```

Comment:
I think a value with both a single quote and a double quote should also be illegal, since quotes can't be escaped.